x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/DocVectorTests.java

```diff
@@ -62,34 +64,53 @@ public void testNonDecreasingDescendingDocs() {
         docs.close();
     }
 
-    private static int MAX_BUILD_BREAKS_LIMIT = 1391;
+    private static long MAX_BUILD_BREAKS_LIMIT = -1;
+    private static long MAX_SHARD_SEGMENT_DOC_MAP_BREAKS = -1;
+
+    @Before
+    public void calculateBreakLimit() {
```

Comment:
I'd move this to the tests that need them. If the build process doesn't properly release on failure this could leave behind un-released blocks. Better to have that fail in the test, i think.